src/main/java/io/stargate/sgv2/jsonapi/api/v1/TokenFilter.java

```diff
@@ -0,0 +1,27 @@
+package io.stargate.sgv2.jsonapi.api.v1;
+
+import static io.stargate.sgv2.jsonapi.config.constants.HttpConstants.DEPRECATED_AUTHENTICATION_TOKEN_HEADER_NAME;
+
+import io.quarkus.vertx.http.runtime.filters.Filters;
+import io.vertx.core.MultiMap;
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.enterprise.event.Observes;
+
+@RequestScoped
```

Comment:
Should add a JAvadoc to explain logic of why we want to (temporarily) add secondary Header from primary.